ballerina-to-openapi/src/main/java/io/ballerina/openapi/service/mapper/type/RecordTypeMapper.java

```diff
@@ -120,7 +124,13 @@ static List<Schema> mapIncludedRecords(RecordTypeSymbol typeSymbol, Components c
                         .typeDescriptor();
                 Map<String, RecordFieldSymbol> includedRecordFieldMap = includedRecordTypeSymbol.fieldDescriptors();
                 for (Map.Entry<String, RecordFieldSymbol> includedRecordField : includedRecordFieldMap.entrySet()) {
-                    recordFieldMap.remove(includedRecordField.getKey());
+                    RecordFieldSymbol recordFieldSymbol = recordFieldMap.get(includedRecordField.getKey());
+                    RecordFieldSymbol includedRecordFieldValue = includedRecordField.getValue();
+                    boolean isRemovableField = recordFieldSymbol != null && includedRecordFieldValue.typeDescriptor()
+                            .equals(recordFieldSymbol.typeDescriptor()) && !recordFieldSymbol.hasDefaultValue();
```

Comment:
Updated the logic to check the default values

Reply:
The logic will only work for the types in the same package and will not work for the types imported from a different package. In that case, we can ignore this logic and add the default value field again rather than checking the default values. I have suggested this to check whether there are any APIs from semantic model to get the fields defined in a record rather than getting all the effective fields (fields defined in record and fields coming from included record)

Reply:
I don't think there is. cc @dulajdilshan 

Also I'm not entirely sure about this check. Wouldn't this handle the following incorrectly?

```ballerina
type RecA record {|
      string a = "default a";
      string b;
|};

type RecB record {|
      *RecA;
      string a; // makes a required with no default
      int d;
|};
```

Reply:
@MaryamZi , This is handled correctly with this logic update: https://github.com/ballerina-platform/openapi-tools/pull/1781#discussion_r1841059023

Generated OAS for example,

```
RecA:
      required:
      - b
      type: object
      properties:
        a:
          type: string
          default: default a
        b:
          type: string
      additionalProperties: false
    RecB:
      type: object
      allOf:
      - $ref: '#/components/schemas/RecA'
      - required:
        - a
        - d
        type: object
        properties:
          a:
            type: string
          d:
            type: integer
            format: int64
        additionalProperties: false
```
I will update the test cases

@TharmiganK, AFAIK in the semantic model, we receive a typeSymbol, and that typeSymbol only contains the API for checking whether it has a default value. However, we cannot obtain the default value information from it. This was discussed during the meeting some time ago. @dulajdilshan will be able to verify if they know have an API we can use for getting value?

Reply:
> However, we cannot obtain the default value information from it.

Semantic API doesn't support getting default value information regarding fields in the records.